src/strands/multiagent/graph.py

```diff
@@ -530,38 +637,61 @@ def _is_node_ready_with_conditions(self, node: GraphNode, completed_batch: list[
                     )
         return False
 
-    async def _execute_node(self, node: GraphNode, invocation_state: dict[str, Any]) -> None:
-        """Execute a single node with error handling and timeout protection."""
+    async def _execute_node(self, node: GraphNode, invocation_state: dict[str, Any]) -> AsyncIterator[dict[str, Any]]:
+        """Execute a single node."""
         # Reset the node's state if reset_on_revisit is enabled and it's being revisited
         if self.reset_on_revisit and node in self.state.completed_nodes:
             logger.debug("node_id=<%s> | resetting node state for revisit", node.node_id)
             node.reset_executor_state()
-            # Remove from completed nodes since we're re-executing it
             self.state.completed_nodes.remove(node)
 
         node.execution_status = Status.EXECUTING
         logger.debug("node_id=<%s> | executing node", node.node_id)
 
+        # Emit node start event
+        start_event = MultiAgentNodeStartEvent(
+            node_id=node.node_id, node_type="agent" if isinstance(node.executor, Agent) else "multiagent"
+        )
+        yield start_event.as_dict()
+
         start_time = time.time()
         try:
             # Build node input from satisfied dependencies
             node_input = self._build_node_input(node)
 
-            # Execute with timeout protection (only if node_timeout is set)
+            # Execute with timeout protection and stream events
             try:
-                # Execute based on node type and create unified NodeResult
                 if isinstance(node.executor, MultiAgentBase):
+                    # For nested multi-agent systems, stream their events and collect result
+                    multi_agent_result = None
                     if self.node_timeout is not None:
```

Comment:
Rather than having two code paths, can `_stream_with_timeout` special_case no timeout to infinite?  

If not, perhaps a conditional like so:

```
events = self._stream_with_timeout(...) if timeout else self.stream_async()
```

So that the if blocks are unified